modality-probe-debug-collector/src/cli.rs

```diff
@@ -0,0 +1,666 @@
+    interval: String,
+
+    /// Output file location
+    #[structopt(short = "o", long = "output-file", parse(from_os_str))]
+    output_path: PathBuf,
+
+    /// Symbols and/or addresses of probes or probe pointers
+    probe_syms: Vec<String>,
+}
+
+/// Turn CLI options into configuration for the collector
+pub(crate) fn config_from_options(options: CLIOptions) -> Result<Config, Box<dyn Error>> {
+    let mut probe_addrs = Vec::new();
+    let mut symbols = Vec::new();
+    // Separate probe/probe pointer addresses and probe symbols
+    for addr_str in options.probe_syms.iter().filter(|s| s.starts_with("0x")) {
+        let addr = u32::from_str_radix(addr_str.trim_start_matches("0x"), 16)?;
+        probe_addrs.push(ProbeAddr::Addr(addr));
+    }
+    for addr_str in options.probe_syms.iter().filter(|s| s.starts_with("*0x")) {
+        let addr = u32::from_str_radix(addr_str.trim_start_matches("*0x"), 16)?;
+        probe_addrs.push(ProbeAddr::PtrAddr(addr));
+    }
+    symbols.extend(
+        options
+            .probe_syms
+            .iter()
+            .filter(|s| !s.starts_with("0x") && !s.starts_with("*0x")),
+    );
+    let mut elf_endianness = None;
+    if let Some(elf_path) = options.elf_path.as_ref() {
+        let mut elf_buf = Vec::new();
+        let elf_file = open_elf(&elf_path, &mut elf_buf)?;
+        // Resolve symbol values
+        for sym in symbols {
+            if let Some((sym_val, _sym_size)) =
+                parse_symbol_info(&elf_file, sym.trim_start_matches('*'))
+            {
+                if sym.starts_with('*') {
+                    probe_addrs.push(ProbeAddr::PtrAddr(sym_val));
+                } else {
+                    probe_addrs.push(ProbeAddr::Addr(sym_val));
+                }
+            } else {
+                return Err(Box::new(OptionsError::new(&format!(
+                    "Could not find symbol value of \"{}\" in given ELF",
+                    sym
+                ))));
+            }
+        }
+        elf_endianness = Some(elf_file.as_ref().header.endianness());
+    } else if !symbols.is_empty() {
+        return Err(Box::new(OptionsError::new(
+            "Must specify memory locations of probes or an ELF file to recover symbol values",
+        )));
+    }
+
+    let interval = parse(&options.interval)?;
+
+    Ok(modality_probe_debug_collector::Config {
+        session_id: options.session_id.into(),
+        big_endian: should_use_big_endian(&options, elf_endianness)?,
+        attach_target: options.attach_target,
+        gdb_addr: options.gdb_addr,
+        interval,
+        output_path: options.output_path,
+        probe_addrs,
+    })
+}
+
+/// Open elf file for parsing
+fn open_elf<'a>(path: &PathBuf, elf_buf: &'a mut Vec<u8>) -> Result<Box<Elf<'a>>, Box<dyn Error>> {
+    let mut file = File::open(path)?;
+    file.read_to_end(elf_buf)?;
+    match Elf::parse(elf_buf) {
+        Ok(elf) => Ok(Box::new(elf)),
```

Comment:
IIUC, we don't need to be boxing this Elf object, and can instead just return it directly (with appropriate update to function signature).